HACKER/HACKER/Sources/ViewControllers/ShoppingRankingViewController/Cells/ShoppingCollectionViewCell.swift

```diff
@@ -0,0 +1,108 @@
+  }
+}
+// MARK: - Extensions
+extension ShoppingCollectionViewCell {
+  func register() {
+    self.myFriendsCollectionView.register(AddFriendCollectionViewCell.self, forCellWithReuseIdentifier: AddFriendCollectionViewCell.identifier)
+    self.myFriendsCollectionView.register(MyFriendsCollectionViewCell.self, forCellWithReuseIdentifier: MyFriendsCollectionViewCell.identifier)
+  }
+  func attribute() {
+    self.myFriendsCollectionView.delegate = self
+    self.myFriendsCollectionView.dataSource = self
+  }
+  func layout() {
+    layoutMyFriendsCollectionView()
+  }
+  func layoutMyFriendsCollectionView() {
+    self.contentView.add(myFriendsCollectionView) {
+      $0.backgroundColor = .hackerWhite
+      $0.isUserInteractionEnabled = true
+      $0.showsVerticalScrollIndicator = false
+      $0.snp.makeConstraints { make in
+        make.top.equalToSuperview().offset(10)
+        make.leading.equalToSuperview().offset(24)
+        make.trailing.equalToSuperview().offset(-24)
+        make.bottom.equalToSuperview()
+      }
+    }
+  }
+  func setupNewFriend() {
+    let addFriendVC = AddFriendViewController()
+    self.parentViewController?.navigationController?.pushViewController(addFriendVC, animated: false)
+  }
+}
+// MARK: - UICollectionViewDataSource
+extension ShoppingCollectionViewCell: UICollectionViewDataSource {
+  func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+    return 30
+  }
+  func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+    guard let addCell = collectionView.dequeueReusableCell(withReuseIdentifier: AddFriendCollectionViewCell.identifier, for: indexPath) as? AddFriendCollectionViewCell else { return UICollectionViewCell() }
+    guard let friendCell = collectionView.dequeueReusableCell(withReuseIdentifier: MyFriendsCollectionViewCell.identifier, for: indexPath) as? MyFriendsCollectionViewCell else { return UICollectionViewCell() }
+    
+    if indexPath.item == 0 {
+      addCell.awakeFromNib()
+      return addCell
+    } else {
+      friendCell.awakeFromNib()
+      return friendCell
+    }
+  }
+  func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+    if indexPath.item == 0 {
+      self.setupNewFriend()
+    }
+  }
+}
+
+// MARK: - UICollectionViewDelegateFlowLayout
+extension ShoppingCollectionViewCell: UICollectionViewDelegateFlowLayout {
+  func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+    let cellWidth = (collectionView.frame.width-24)/3
+    let cellHeight = (cellWidth)*1.38
+    return CGSize(width: cellWidth, height: cellHeight)
+  }
+  func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+    return UIEdgeInsets.zero
```

Comment:
위에서 말했던 컬렉션뷰 Inset 바꾸는 곳!
top말고 bottom에도 inset을 준 이유는 컬렉션뷰 셀 가장 아래까지 스크롤 했을 때 걔가 너무 바닥에 붙어있지 않아 보이게 하기 위해서야

```suggestion
    return UIEdgeInsets(top: 10, left: 0, bottom: 10, right: 0)
```